lib/resource.go

```diff
@@ -60,8 +62,8 @@ func (u *URL) HeadersMap() (headers map[string]string) {
 	return
 }
 
-// HandleGetResponse handles a url GET
-func (u *URL) HandleGetResponse(started time.Time, res *http.Response, recordHeaders bool) (err error) {
+// HandleResponse populates a resource based on an HTTP response
+func (u *Resource) HandleResponse(started time.Time, res *http.Response, recordHeaders bool) (err error) {
```

Comment:
I feel like we’ll eventually want to move this and some of the other setter methods to the actual worker, so the Resource type can easily be used in handlers, etc.

A non-go process (e.g. Node.js using Puppeteer) wouldn't use these implementations and, in that scenario, none of the go parts would either, since they are only reading the resource).

Reply:
Totally agreed. This feels like a near-future PR. This method has always made me feel icky.